### ISSUE TITLE
No retroactive application of place-count when changing to a different resource-group

### ISSUE THREAD

**Issue by candlerb** [LINBIT/linstor-server#222]
I noticed an apparent inconsistency:

1. If you create a resource under a resource-group with place-count=2, and then modify the resource-group itself to place-count=3, then additional resources are automatically deployed for the child resource-definitions
2. If you create a resource under a resource-group with place-count=2, and then change its resource-definition to point to a different resource-group which has a place-count=3, no resources are deployed (you have to use "auto-place")

Case 1 is working as documented [here](https://www.linbit.com/drbd-user-guide/linstor-guide-1_0-en/#s-linstor-resource-groups):

> resource groups are like templates that define characteristics of resources created from them. Changes to these pseudo templates will be applied to all resources that were created from the resource group, retroactively.

However if case 2 is the intended behaviour, then I think it would be worth mentioning here as well.

Illustration of case 2:

```
root@node1:~# linstor resource-group list
╭─────────────────────────────────────────────────────────────────╮
┊ ResourceGroup ┊ SelectFilter             ┊ VlmNrs ┊ Description ┊
╞═════════════════════════════════════════════════════════════════╡
┊ DfltRscGrp    ┊ PlaceCount: 2            ┊        ┊             ┊
╞┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄╡
┊ my_ssd_group  ┊ PlaceCount: 2            ┊ 0      ┊             ┊
┊               ┊ StoragePool(s): pool_ssd ┊        ┊             ┊
╞┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄╡
┊ ssd_crit      ┊ PlaceCount: 3            ┊ 0      ┊             ┊
┊               ┊ StoragePool(s): pool_ssd ┊        ┊             ┊
╞┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄╡
┊ ssd_std       ┊ PlaceCount: 2            ┊ 0      ┊             ┊
┊               ┊ StoragePool(s): pool_ssd ┊        ┊             ┊
╰─────────────────────────────────────────────────────────────────╯

root@node1:~# linstor resource-group spawn-resources ssd_std bar 2G
...
root@node1:~# linstor resource list -r bar
╭────────────────────────────────────────────────────────────────────────────────────────╮
┊ ResourceName ┊ Node  ┊ Port ┊ Usage  ┊ Conns ┊             State ┊ CreatedOn           ┊
╞════════════════════════════════════════════════════════════════════════════════════════╡
┊ bar          ┊ node1 ┊ 7001 ┊ Unused ┊ Ok    ┊ SyncTarget(0.77%) ┊ 2021-02-15 11:24:59 ┊
┊ bar          ┊ node2 ┊ 7001 ┊ Unused ┊ Ok    ┊          UpToDate ┊ 2021-02-15 11:24:59 ┊
╰────────────────────────────────────────────────────────────────────────────────────────╯
root@node1:~# linstor resource-definition modify --resource-group ssd_crit bar
SUCCESS:
...
root@node1:~# linstor resource list -r bar
╭────────────────────────────────────────────────────────────────────────────────────────╮
┊ ResourceName ┊ Node  ┊ Port ┊ Usage  ┊ Conns ┊             State ┊ CreatedOn           ┊
╞════════════════════════════════════════════════════════════════════════════════════════╡
┊ bar          ┊ node1 ┊ 7001 ┊ Unused ┊ Ok    ┊ SyncTarget(4.40%) ┊ 2021-02-15 11:24:59 ┊
┊ bar          ┊ node2 ┊ 7001 ┊ Unused ┊ Ok    ┊          UpToDate ┊ 2021-02-15 11:24:59 ┊
╰────────────────────────────────────────────────────────────────────────────────────────╯
root@node1:~# linstor resource-definition auto-place bar
...
root@node1:~# linstor resource list -r bar
╭────────────────────────────────────────────────────────────────────────────────────────╮
┊ ResourceName ┊ Node  ┊ Port ┊ Usage  ┊ Conns ┊             State ┊ CreatedOn           ┊
╞════════════════════════════════════════════════════════════════════════════════════════╡
┊ bar          ┊ node1 ┊ 7001 ┊ Unused ┊ Ok    ┊ SyncTarget(7.14%) ┊ 2021-02-15 11:24:59 ┊
┊ bar          ┊ node2 ┊ 7001 ┊ Unused ┊ Ok    ┊          UpToDate ┊ 2021-02-15 11:24:59 ┊
┊ bar          ┊ node3 ┊ 7001 ┊ Unused ┊ Ok    ┊ SyncTarget(0.61%) ┊ 2021-02-15 11:24:58 ┊
╰────────────────────────────────────────────────────────────────────────────────────────╯
```

**Comment by candlerb** [comment]
This *doesn't* appear to have been fixed in linstor controller/satellite 1.12.0:

```
linstor resource-group create --storage-pool pool_ssd --place-count 3 ssd_crit
linstor resource-group create --storage-pool pool_ssd --place-count 2 ssd_std
linstor volume-group create ssd_crit
linstor volume-group create ssd_std
linstor resource-group spawn-resources ssd_std bar 2G
```

```
root@node1:~# linstor resource list -r bar
╭────────────────────────────────────────────────────────────────────────────────────────╮
┊ ResourceName ┊ Node  ┊ Port ┊ Usage  ┊ Conns ┊             State ┊ CreatedOn           ┊
╞════════════════════════════════════════════════════════════════════════════════════════╡
┊ bar          ┊ node1 ┊ 7001 ┊ Unused ┊ Ok    ┊ SyncTarget(4.96%) ┊ 2021-04-30 08:51:21 ┊
┊ bar          ┊ node3 ┊ 7001 ┊ Unused ┊ Ok    ┊          UpToDate ┊ 2021-04-30 08:51:21 ┊
╰────────────────────────────────────────────────────────────────────────────────────────╯
root@node1:~# linstor resource-definition modify --resource-group ssd_crit bar
SUCCESS:
Description:
    Resource definition 'bar' modified.
Details:
    Resource definition 'bar' UUID is: 86380273-f059-4688-82cb-4d355b513dd2
SUCCESS:
    (node2) Resource 'bar' [DRBD] adjusted.
SUCCESS:
    (node3) Resource 'bar' [DRBD] adjusted.
SUCCESS:
    (node1) Resource 'bar' [DRBD] adjusted.
root@node1:~# linstor resource list -r bar
╭─────────────────────────────────────────────────────────────────────────────────────────╮
┊ ResourceName ┊ Node  ┊ Port ┊ Usage  ┊ Conns ┊              State ┊ CreatedOn           ┊
╞═════════════════════════════════════════════════════════════════════════════════════════╡
┊ bar          ┊ node1 ┊ 7001 ┊ Unused ┊ Ok    ┊ SyncTarget(11.56%) ┊ 2021-04-30 08:51:21 ┊
┊ bar          ┊ node3 ┊ 7001 ┊ Unused ┊ Ok    ┊           UpToDate ┊ 2021-04-30 08:51:21 ┊
╰─────────────────────────────────────────────────────────────────────────────────────────╯
root@node1:~# linstor resource-definition list
╭─────────────────────────────────────────────╮
┊ ResourceName ┊ Port ┊ ResourceGroup ┊ State ┊
╞═════════════════════════════════════════════╡
┊ bar          ┊ 7001 ┊ ssd_crit      ┊ ok    ┊
┊ my_ssd_res   ┊ 7000 ┊ my_ssd_group  ┊ ok    ┊
╰─────────────────────────────────────────────╯
```

That is, although resource `bar` has now been moved to group `ssd_crit`, it still only has the two replicas created when it was in `ssd_std`.

I was going to retest with 1.12.1, but currently cannot due to #229

**Comment by ghernadi** [comment]
Hi, sorry - that must have slipped through. I added the new behavior, but forgot about scenarios with tiebreaking resources. To make sure this does not happen again I added tests 

<details><summary>with auto-tiebreaker enabled and</summary>

```
linstor c sp DrbdOptions/auto-add-quorum-tiebreaker True
linstor n c bravo
linstor n c charlie
linstor n c delta
linstor sp c lvm bravo lvmpool scratch
linstor sp c lvm charlie lvmpool scratch
linstor sp c lvm delta lvmpool scratch
linstor rg c rg2 --place-count 2
linstor vg c rg2
linstor rg c rg3 --place-count 3
linstor vg c rg3
linstor rg spawn rg2 rsc 100m
sleep 10.0s     # give DRBD some time to reach UpToDate state
linstor --no-utf8 --no-color r l -a
+-----------------------------------------------------------------------------------+
| ResourceName | Node    | Port | Usage  | Conns |      State | CreatedOn           |
|===================================================================================|
| rsc          | bravo   | 7000 | Unused | Ok    |   UpToDate | 2021-05-04 14:02:02 |
| rsc          | charlie | 7000 | Unused | Ok    |   UpToDate | 2021-05-04 14:02:02 |
| rsc          | delta   | 7000 | Unused | Ok    | TieBreaker | 2021-05-04 14:01:58 |
+-----------------------------------------------------------------------------------+

linstor rd m --resource-group rg3 rsc
sleep 10.0s     # give DRBD some time to reach UpToDate state
linstor --no-utf8 --no-color r l -a
+---------------------------------------------------------------------------------+
| ResourceName | Node    | Port | Usage  | Conns |    State | CreatedOn           |
|=================================================================================|
| rsc          | bravo   | 7000 | Unused | Ok    | UpToDate | 2021-05-04 14:02:02 |
| rsc          | charlie | 7000 | Unused | Ok    | UpToDate | 2021-05-04 14:02:02 |
| rsc          | delta   | 7000 | Unused | Ok    | UpToDate | 2021-05-04 14:01:58 |
+---------------------------------------------------------------------------------+
```
</details>
<details><summary>without</summary>

```

linstor c sp DrbdOptions/auto-add-quorum-tiebreaker False
linstor n c bravo
linstor n c charlie
linstor n c delta
linstor sp c lvm bravo lvmpool scratch
linstor sp c lvm charlie lvmpool scratch
linstor sp c lvm delta lvmpool scratch
linstor rg c rg2 --place-count 2
linstor vg c rg2
linstor rg c rg3 --place-count 3
linstor vg c rg3
linstor rg spawn rg2 rsc 100m
sleep 10.0s     # give DRBD some time to reach UpToDate state
linstor --no-utf8 --no-color r l -a
+---------------------------------------------------------------------------------+
| ResourceName | Node    | Port | Usage  | Conns |    State | CreatedOn           |
|=================================================================================|
| rsc          | bravo   | 7000 | Unused | Ok    | UpToDate | 2021-05-04 14:02:55 |
| rsc          | charlie | 7000 | Unused | Ok    | UpToDate | 2021-05-04 14:02:55 |
+---------------------------------------------------------------------------------+

linstor rd m --resource-group rg3 rsc
sleep 10.0s     # give DRBD some time to reach UpToDate state
linstor --no-utf8 --no-color r l -a
+---------------------------------------------------------------------------------+
| ResourceName | Node    | Port | Usage  | Conns |    State | CreatedOn           |
|=================================================================================|
| rsc          | bravo   | 7000 | Unused | Ok    | UpToDate | 2021-05-04 14:02:55 |
| rsc          | charlie | 7000 | Unused | Ok    | UpToDate | 2021-05-04 14:02:55 |
| rsc          | delta   | 7000 | Unused | Ok    | UpToDate | 2021-05-04 14:03:08 |
+---------------------------------------------------------------------------------+
```
</details>

**Comment by rp-** [comment]
should be fixed in v1.12.3
